src/base_home.cpp

```diff
@@ -0,0 +1,94 @@
+#include "base_home.h"
+#include "submap.h"
+#include <unordered_map>
+#include "mapdata.h"
+#include "map.h"
+
+base_home::base(submap &base_map,const tripoint &coreloc){
+    base::define_base_area(base_map, coreloc);
+    base_map.spawns.clear();//remove spawns so nothing apears in base just outside it.
+    base_level = 1;
+    food_ration_lv = ammo_ration_lv = med_ration_lv = 0; //no rationing to start
+}
+
+/**
+ * Check if location has a base specific flag.
+ */
+bool base_home::has_base_flag(const tripoint &p, base_area_flag flag)
+{
+    return(baflag[p.x][p.y])
+}
+
+/**
+ * Populates data members containing information read from the map.
+ * 
+ * CAUTION: WILL RESET THE BASE!!! (mostly)
+ * TODO: Populate baflag with flags coresponding to the structure containing the command desk.
+ * TODO: Populate storage_open with locations of storage furnature (lockers etc.)
+ * TODO: Populate bunks with locations of sleeping areas (beds, cots, etc.)
+ */
+void base_home::define_base_area(const submap &base_map, const tripoint &coreloc)
+{
+    //for easy reference
+    int x = coreloc.x;
+    int y = coreloc.y;
+
+    //TODO:  Re-write. Previous implementation wasn't going to work. (immpossible to distinguish between underground and thick rock wall)
```

Comment:
Scanning area isn't easy, but `start_location.cpp` has an implementation of flood fill that would work to find all the reachable area from current point.